### PR TITLE
[protocol change 52] Increasing max_gas_burnt to 300

### PR DESCRIPTION
**Pull request by mm-near**
Increasing max_gas_burnt to 300

# Testing and QA
Describe the testing plan for this protocol and why you are confident that it is ready to be stabilized.

# Checklist
- [ ] Link to nightly nayduck run (`./scripts/nayduck.py`, [docs](https://github.com/near/nearcore/blob/master/nightly/README.md#scheduling-a-run)):  https://nayduck.near.org/
- [ ] Update CHANGELOG.md to include this protocol feature in the `Unreleased` section.

### CODE REVIEW

File: core/primitives/src/runtime/config_store.rs [hunk 1]
```diff
@@ -20,6 +20,8 @@ static CONFIGS: &[(ProtocolVersion, &[u8])] = &[
     (48, include_config!("48.json")),
     (49, include_config!("49.json")),
     (50, include_config!("50.json")),
+    // max_gas_burnt increased to 300
```
Comment (matklad) [inline core/primitives/src/runtime/config_store.rs:23]:
```suggestion
    // max_gas_burnt increased to 300 TGas
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Changelog
 
 ## [unreleased]
+* increasing max_gas_burnt to 300, allowing larger transactions.
```
Comment (matklad) [inline CHANGELOG.md:4]:
```suggestion
* increasing max_gas_burnt from 200TGas to 300Tgas, allowing larger transactions.
```

File: core/primitives/res/runtime_configs/52.json [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+            "max_promises_per_function_call_action": 1024,
+            "max_number_input_data_dependencies": 128,
+            "max_functions_number_per_contract": 10000
+        }
+    },
+    "account_creation_config": {
+        "min_allowed_top_level_account_length": 32,
+        "registrar_account_id": "registrar"
+    }
+}
```
Comment (matklad) [inline core/primitives/res/runtime_configs/52.json:193]: Please minimize the diff with the previous config:

```
$ diff core/primitives/res/runtime_configs/52.json core/primitives/res/runtime_configs/50.json | wc -l
386
```

**Review by matklad — COMMENTED** [review]
Do we want to make this insta-stable? (ie, without protocol_features business?)

File: core/primitives/res/runtime_configs/52.json (hunk 3 above)
Comment (matklad) [reply core/primitives/res/runtime_configs/52.json:193]: (alternatively, prepend a commint which reformats previous configs to 4 spaces)

Comment (mm-near) [reply core/primitives/res/runtime_configs/52.json]: good point - fixed.

**Review by matklad — APPROVED** [review]
LGTM!

Still not sure whether we should bother with a protocol_feature for this one, I don't think we decided one way or another. 

My gut feeling is the process where we don't do protocol feature for purely config changes is OK.

File: core/primitives/src/version.rs [hunk 4]
```diff
@@ -155,7 +155,7 @@ pub const PEER_MIN_ALLOWED_PROTOCOL_VERSION: ProtocolVersion = MAIN_NET_PROTOCOL
 /// Current protocol version used on the main net.
 /// Some features (e. g. FixStorageUsage) require that there is at least one epoch with exactly
 /// the corresponding version
-const MAIN_NET_PROTOCOL_VERSION: ProtocolVersion = 51;
+const MAIN_NET_PROTOCOL_VERSION: ProtocolVersion = 52;
```
Comment (bowenwang1996) [inline core/primitives/src/version.rs:158]: Unrelated to this PR but the naming confused me: why mainnet specifically?

**Review by bowenwang1996 — CHANGES_REQUESTED** [review]
This *is* a protocol change. If we do it the way it is done right now, if someone sends a transaction that consumes 201Tgas in a single call during the upgrade process, the network will split

**Review by bowenwang1996 — APPROVED** [review]
Nvm I got confused

File: core/primitives/src/version.rs (hunk 4 above)
Comment (matklad) [reply core/primitives/src/version.rs:158]: Was introduced in https://github.com/near/nearcore/pull/5610/. I *think* it's indeed better to rename it to `STABLE_PROTOCOL_VERSION`. We need this for `PEER_MIN_ALLOWED_PROTOCOL_VERSION`, to allow nightly peers to connect to mainnet I think.

**Comment by mm-near** [comment]
our tests are failing with:

[2022-02-01 13:21:31] INFO: Latest mainnet release is 1.23.1
  | Traceback (most recent call last):
  | File "tests/sanity/upgradable.py", line 179, in <module>
  | main()
  | File "tests/sanity/upgradable.py", line 174, in main
  | test_protocol_versions()
  | File "tests/sanity/upgradable.py", line 54, in test_protocol_versions
  | f'If changed, protocol version of a new release can increase by at '
  | AssertionError: If changed, protocol version of a new release can increase by at most one.
  | Got protocol 50 version in release 1.23.1
  | but protocol 52 version in trunk

**Comment by matklad** [comment]
cc https://github.com/near/nearcore/pull/5913/files#r773203306

File: core/primitives/src/version.rs (hunk 4 above)
Comment (bowenwang1996) [reply core/primitives/src/version.rs:158]: Agree that `STABLE_PROTOCOL_VERSION` makes a lot more sense

**Pull request merged**